presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java

```diff
@@ -314,12 +389,60 @@ protected String getTableSchemaName(ResultSet resultSet)
     @Override
     public WriteMapping toWriteMapping(ConnectorSession session, Type type)
     {
-        if (DOUBLE.equals(type)) {
-            return WriteMapping.doubleMapping("double", doubleWriteFunction());
+        if (type == BOOLEAN) {
+            return WriteMapping.booleanMapping("boolean", booleanWriteFunction());
         }
-        if (REAL.equals(type)) {
+
+        if (type == TINYINT) {
+            return WriteMapping.longMapping("tinyint", tinyintWriteFunction());
+        }
+        if (type == SMALLINT) {
+            return WriteMapping.longMapping("smallint", smallintWriteFunction());
+        }
+        if (type == INTEGER) {
+            return WriteMapping.longMapping("integer", integerWriteFunction());
+        }
+        if (type == BIGINT) {
+            return WriteMapping.longMapping("bigint", bigintWriteFunction());
+        }
+        if (type == REAL) {
             return WriteMapping.longMapping("float", realWriteFunction());
         }
+        if (type == DOUBLE) {
+            return WriteMapping.doubleMapping("double", doubleWriteFunction());
+        }
+
+        if (type instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) type;
+            String dataType = format("decimal(%s, %s)", decimalType.getPrecision(), decimalType.getScale());
+            if (decimalType.isShort()) {
+                return WriteMapping.longMapping(dataType, shortDecimalWriteFunction(decimalType));
+            }
+            return WriteMapping.sliceMapping(dataType, longDecimalWriteFunction(decimalType));
+        }
+
+        if (type instanceof CharType) {
+            return WriteMapping.sliceMapping("char(" + ((CharType) type).getLength() + ")", charWriteFunction());
+        }
+        if (type instanceof VarcharType) {
+            VarcharType varcharType = (VarcharType) type;
+            String dataType;
+            if (varcharType.isUnbounded()) {
+                // TODO does Phoenix support unparameterized varchar?
```

Comment:
issue?

Reply:
i need to remove this TODO, per https://github.com/prestosql/presto/pull/6121#discussion_r532261224